base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs

```diff
@@ -336,7 +344,11 @@ where KM: TransactionKeyManagerInterface
             .fee()
             .calculate(fee_per_gram, 0, 0, 1, change_features_and_scripts_size);
         // Subtract with a check on going negative
-        let total_input_value = total_to_self + total_amount + fee_without_change;
+        let total_input_value = [total_to_self, total_amount, fee_without_change]
+            .iter()
+            .fold(Ok(MicroMinotari::zero()), |acc, x| {
+                acc?.checked_add(*x).ok_or("Total input value overflow")
```

Comment:
Strange it needs to be dereferenced here `checked_add(*x)`

Reply:
`.checked_add(` is not implemented for a reference value

Reply:
is it possible to add that implementation?

Reply:
We could have 
``` rust
    pub fn checked_add(self, v: MicroMinotari) -> Option<MicroMinotari> {
        self.as_u64().checked_add(v.as_u64()).map(Into::into)
    }

    pub fn checked_add_ref(self, v: &MicroMinotari) -> Option<MicroMinotari> {
        self.as_u64().checked_add(v.as_u64()).map(Into::into)
    }
```
as
``` rust
new_val = val.checked_add_ref(x).ok_or("Total input value overflow");
```
vs.
``` rust
new_val = val.checked_add(*x).ok_or("Total input value overflow")
```
but I do not see the benefit of the additional function when we can just dereference when needed.

Reply:
Ok, please disregard my previous comment, I implemented `AsRef` for those methods to facilitate the overloading